webapp/views.py

```diff
@@ -19,6 +30,197 @@ def get_cve(cve_id):
+            "SELECT unnest(enum_range(NULL::statuses));"
+        ).fetchall()
+        all_statuses = ["".join(s) for s in raw_all_statuses]
+
+        clean_versions = [
+            (
+                [version]
+                if version not in ["", "current"]
+                else [r.codename for r in releases]
+            )
+            for version in versions
+        ]
+
+        clean_statuses = [
+            ([status] if status != "" else all_statuses) for status in statuses
+        ]
+
+    # query cves by filters
+    cves_query = db_session.query(
+        CVE, func.count("*").over().label("total")
+    ).filter(CVE.status == "active")
+
+    if priority:
+        cves_query = cves_query.filter(CVE.priority == priority)
+
+    if query:
+        cves_query = cves_query.filter(
+            or_(
+                CVE.description.ilike(f"%{query}%"),
+                CVE.ubuntu_description.ilike(f"%{query}%"),
+            )
+        )
+
+    parameters = []
+    if package:
+        parameters.append(Status.package_name == package)
+
+    if component:
+        parameters.append(Status.component == component)
+
+    if should_filter_by_version_and_status:
+        conditions = []
+        for key, version in enumerate(clean_versions):
+            conditions.append(
+                and_(
+                    Status.release_codename.in_(version),
+                    Status.status.in_(clean_statuses[key]),
+                )
+            )
+
+        parameters.append(or_(*[c for c in conditions]))
+
+        conditions = []
+        for key, version in enumerate(clean_versions):
+            sub_conditions = [
+                Status.release_codename.in_(version),
+                Status.status.in_(clean_statuses[key]),
+                CVE.id == Status.cve_id,
+            ]
+
+            if package:
+                sub_conditions.append(Status.package_name == package)
+
+            if component:
+                sub_conditions.append(Status.component == component)
+
+            condition = Package.statuses.any(
+                and_(*[sc for sc in sub_conditions])
+            )
+
+            conditions.append(condition)
+
+        parameters.append(Status.package.has(and_(*[c for c in conditions])))
+    else:
+        parameters.append(Status.status.in_(Status.active_statuses))
+
+    if len(parameters) > 0:
+        cves_query = cves_query.filter(
+            CVE.statuses.any(and_(*[p for p in parameters]))
+        )
+
+    sort = asc if order_by == "oldest" else desc
+
+    cves_query = (
+        cves_query.group_by(CVE.id)
+        .order_by(
+            case(
+                [(CVE.published.is_(None), 1)],
+                else_=0,
+            ),
+            sort(CVE.published),
+        )
+        .limit(limit)
+        .offset(offset)
+        .from_self()
+        .join(CVE.statuses)
+        .options(contains_eager(CVE.statuses))
+    )
+
+    # each result item has
```

Comment:
not really indicative of what is happening here

Reply:
So.. I am no big fan of comments. But after I finished the refactor, I added some to explain a bit the filters that get applied which are fairly complex and I do not see a way to make the easy to understand. 

If you find that they are useless, I can remove them.